alacritty/src/config/mod.rs

```diff
@@ -176,7 +190,11 @@ fn parse_config(contents: &str) -> Result<Config> {
         },
         Ok(config) => {
             print_deprecation_warnings(&config);
-            Ok(config)
+
+            match config.validate() {
+                None => Ok(config),
+                Some(err) => Err(Error::InvalidConfiguration(err)),
+            }
```

Comment:
I don't see why we need to check things both for the CLI and for the Config loading. The CLI values just replace the config, so we should just verify after both have been combined.